### PR TITLE
difference-of-squares: Change expected exception type

### PR DESCRIPTION
**Pull request by balazsbotond**
I think it would be better to use the more specific `ArgumentOutOfRangeException` instead of `ArgumentException` here.

### CODE REVIEW

File: exercises/difference-of-squares/DifferenceOfSquaresTest.cs [hunk 1]
```diff
@@ -77,6 +77,6 @@ public void Test_difference_of_sums_0()
     [Test]
     public void Test_negative_numbers_throw_argument_exception()
```
Comment (robkeim) [inline exercises/difference-of-squares/DifferenceOfSquaresTest.cs]: Could you update the title to say `argument_out_of_range_exception`?

**Comment by robkeim** [comment]
Thanks @balazsbotond for the PR! I've gone ahead and added my input, could you please take a look and update the changes?

This will be a small breaking change for anyone that's already implemented this but doesn't change any of the core problem, so I think it should be fine.

File: exercises/difference-of-squares/DifferenceOfSquaresTest.cs (hunk 1 above)
Comment (balazsbotond) [inline exercises/difference-of-squares/DifferenceOfSquaresTest.cs]: Sure!

**Comment by balazsbotond** [comment]
@robkeim thanks for the review! I've renamed the test.

**Comment by robkeim** [comment]
Merging this, thanks @balazsbotond!

**Pull request merged**